src/include/daos_srv/vos_types.h

```diff
@@ -32,6 +32,17 @@
 #include <daos/dtx.h>
 #include <daos/checksum.h>
 
+struct dtx_rsrvd_scm {
+	uint32_t		drs_cnt;
+	uint32_t		drs_at;
+	void			*drs_ptr;
+};
+
+struct dtx_rsrvd_uint {
+	struct dtx_rsrvd_scm	dru_scm;
+	d_list_t		dru_nvme;
+};
```

Comment:
Suggestion: It's better to not exposing these VOS internal data structure in vos_types.h, so that you don't have to create a dup structure of dtx_rsrvd_scm (which is identical to vos_rsrvd_scm).

You could put an opaque "vos_rsrvd_unit *" in dtx handle, or make sure the buffer size is large enough to contain the "vos_rsrvd_unit" if you want to make inline.

Reply:
We do not export VOS internal data structure. If we can expose that, then the patch will directly use "vos_rervd_scm".

Reply:
Just because we cannot expose the "vos_rsrvd_scm", we have to define another "dtx_rsrvd_scm" out side of VOS, and it  check whether it is exactly matches "vos_rsrvd_scm" inside VOS.

Reply:
If we really want to avoid "dtx_rsrvd_scm", then the DTX handle need to reference the "vos_rsrvd_scm" pointer inside VOS. But such way will much change current vos_aggregation and vos_io_context related logic, just as I did in former implementation, but you did not like such way. That is why we have the new version.